### DISCUSSION TITLE
Undici safe for the browser?

### DISCUSSION THREAD

**Discussion by bcomnes** [nodejs/undici#1016] [Q&A] [answered]
Hey, just reading through the docs, but not totally clear on this detail:

Can undici be used in the browser? Or is there a compatibility layer that uses undici in node, and fetch in browser or something?

**Comment by szmarczak** [comment, marked as answer]
`undici` is written only for Node.js

**Comment by bcomnes** [comment]
Anyone know of a universal request library wrapper that works in the browser and node, and uses undici when in node?

**Reply by mcollina** [reply]
Well, undici ships with fetch in Node v16, so it might just be about plumbing the polyfill in, or installing undici fetch as a global.

**Reply by bcomnes** [reply]
Oh, I see, so import fetch from the window or undici maybe.  I'll play around with that.

**Reply by bcomnes** [reply]
I wrote a small wrapper that provides `window.fetch` in the browser and `undici.fetch` in Node in case anyone wants that.

https://github.com/bcomnes/fetch-undici
